test/integration/src/test/java/io/pravega/test/integration/RestoreBackUpDataRecoveryTest.java

```diff
@@ -175,30 +159,27 @@ protected int getThreadPoolSize() {
         return 100;
     }
 
-    BookKeeperStarter setUpNewBK(int instanceId) throws Exception {
-        return new BookKeeperStarter(instanceId);
-    }
-
     /**
      * Sets up a new BookKeeper & ZooKeeper.
      */
-    private static class BookKeeperStarter implements AutoCloseable {
-        private final int bookieCount = 1;
-        private AtomicReference<BookKeeperConfig> bkConfig = new AtomicReference<>();
-        private AtomicReference<CuratorFramework> zkClient = new AtomicReference<>();
-        private BookKeeperServiceRunner bookKeeperServiceRunner;
-        private AtomicReference<BookKeeperServiceRunner> bkService = new AtomicReference<>();
-        private int bkPort;
-
-        BookKeeperStarter(int instanceId) throws Exception {
+    private static class BookKeeperRunner implements AutoCloseable {
+        private final int bkPort;
+        private final BookKeeperServiceRunner bookKeeperServiceRunner;
+        private final AtomicReference<BookKeeperConfig> bkConfig = new AtomicReference<>();
+        private final AtomicReference<CuratorFramework> zkClient = new AtomicReference<>();
+        private final AtomicReference<BookKeeperServiceRunner> bkService = new AtomicReference<>();
+
+        BookKeeperRunner(int instanceId, int bookieCount) throws Exception {
```

Comment:
what happens when there are exceptions here ?
Let's make sure test exits cleanly.
I suggest we use 
```
try {
} catch (Exception e) {
   this.close()
}
```

Reply:
this.close() also throws `Exception`.

Reply:
Which is okay. Goal is to call close so that we correctly call close  on bk and process.

This also means we need to fix close so that it doesn't leak.

Reply:
Ok. Calling close if bookKeeper start fails.